tests/test_username_password.py

```diff
@@ -425,6 +425,15 @@ def test_federated_wstrust_unknown_token_type(self):
             #assert
             self.assertEqual('Unsuccessful RSTR.\n\terror code: None\n\tfaultMessage: None', exp.args[0])
 
+    def test_parse_id_token_with_unicode(self):
+        client = self.create_oauth2_client_stub('https://foo/ba.onmicrosoft.com', None, None)
+        encoded_token = 'eyJ0eXAiOiJKV1QiLCJhbGciOiJub25lIn0.eyJhdWQiOiIwNGIwNzc5NS04ZGRiLTQ2MWEtYmJlZS0wMmY5ZTFiZjdiNDYiLCJpc3MiOiJodHRwczovL3N0cy53aW5kb3dzLm5ldC81NDgyNmIyMi0zOGQ2LTRmYjItYmFkOS1iN2I5M2EzZTljNWEvIiwiaWF0IjoxNDc3NDM0MDE1LCJuYmYiOjE0Nzc0MzQwMTUsImV4cCI6MTQ3NzQzNzkxNSwiYW1yIjpbInB3ZCJdLCJmYW1pbHlfbmFtZSI6Ikd1aW5lYmVydGnDqHJlIiwiZ2l2ZW5fbmFtZSI6IkJlbmphbWluIiwiaXBhZGRyIjoiMTY3LjIyMC4wLjEzMCIsIm5hbWUiOiJCZW5qYW1pbiBHdWluZWJlcnRpw6hyZSIsIm9pZCI6ImIxNzIzOGNiLWRhYTEtNGU1Mi1hNjBkLTFmY2Q2M2M2MGU5NCIsInN1YiI6IjlRc0dod0tvekpYZjZzZ0c2Uk5qazVJcXlXMXJMUXhmX0NFNWltbHdBRFUiLCJ0aWQiOiI1NDgyNmIyMi0zOGQ2LTRmYjItYmFkOS1iN2I5M2EzZTljNWEiLCJ1bmlxdWVfbmFtZSI6ImJlbmd1aUBBenVyZVNES1RlYW0ub25taWNyb3NvZnQuY29tIiwidXBuIjoiYmVuZ3VpQEF6dXJlU0RLVGVhbS5vbm1pY3Jvc29mdC5jb20iLCJ2ZXIiOiIxLjAifQ.'
```

Comment:
I can do that. I thought we should try to use real data if it doesn't add up the cose